packages/compiler/core/checker.ts

```diff
@@ -1748,18 +1747,33 @@ export function createChecker(program: Program): Checker {
     if (!sym) return sym;
 
     if (sym.flags & SymbolFlags.DuplicateUsing) {
-      reportAmbiguousIdentifier(node, [...((table.duplicates.get(sym) as any) ?? [])]);
+      const duplicates: { sym: Sym; isGlobal: boolean }[] = [];
+      for (const symbol of [...(table.duplicates.get(sym) ?? [])]) {
+        duplicates.push({ sym: symbol, isGlobal: false });
+      }
+      reportAmbiguousIdentifier(node, duplicates);
       return sym;
     }
     return getMergedSymbol(sym);
   }
 
-  function reportAmbiguousIdentifier(node: IdentifierNode, symbols: Sym[]) {
-    const duplicateNames = symbols.map(getFullyQualifiedSymbolName).join(", ");
+  function reportAmbiguousIdentifier(
+    node: IdentifierNode,
+    symbols: { sym: Sym; isGlobal: boolean }[]
+  ) {
+    let duplicateNames: string[] = [];
+    for (const data of symbols) {
```

Comment:
nit: still using a map instead of for each + push would be a tiny bit cleaner but whatever you prefer

Reply:
Pushed the change.